native/libcst/src/tokenizer/whitespace_parser.rs

```diff
@@ -376,13 +374,18 @@ pub fn parse_simple_whitespace<'a>(
     state: &mut State,
 ) -> Result<SimpleWhitespace<'a>> {
     let capture_ws = |line, col| -> Result<&'a str> {
-        let x = config.get_line_after_column(line, col);
-        let x = x?;
-        Ok(SIMPLE_WHITESPACE_RE.with(|r| {
-            r.find(x)
-                .expect("SIMPLE_WHITESPACE_RE supports 0-length matches, so it must always match")
-                .as_str()
-        }))
+        let line = config.get_line_after_column(line, col)?;
+        let bytes = line.as_bytes();
+        let mut idx = 0;
+        while idx < bytes.len() {
```

Comment:
An abstraction that worked really well for Ruff (and Rust) is to use the following `Cursor` abstraction to lex strings. It avoids the need for these manual `match` statements. 

https://github.com/astral-sh/ruff/blob/4d49d5e8451277f8159a30b7da187626d3a75494/crates/ruff_python_parser/src/lexer/cursor.rs#L14-L13